packager/rbi_gen.cc

```diff
@@ -327,7 +327,7 @@ class RBIExporter final {
         if (methodData->flags.isPrivate) {
             if (methodData->owner.data(gs)->isSingletonClass(gs)) {
                 visibility = "private_class_method ";
-            } else {
+            } else if (methodData->name != core::Names::initialize()) {
```

Comment:
This is the main change, and is mimicked by the Ruby VM:

https://github.com/ruby/ruby/blob/6847ee089d7655b2a0eea4fee3133aeacd4cc7cc/vm_method.c#L546-L557

(They also have support for `initialize_{copy,clone,dup}` but Sorbet doesn't have any special handling for those yet.)